test/python/test_roles_api.py

```diff
@@ -4,24 +4,714 @@
+            )
+
+        self.assertEqual(context.exception.status, 422)
+
+    # Test cases for getting a role's members
+
+    def test_get_members_200(self):
+        """Test using members query parameter on GET requests
+        Queries group:userGroup role for its members, which should only be dev:user:admin
+        """
+        response = self.api.get_role_with_http_info(
+            self.account,
+            'group',
+            'userGroup',
+            members=''
+        )
+
+        target_response = [
+            {
+                'admin_option': True,
+                'ownership': True,
+                'role': self.GROUP_ID,
+                'member': self.ADMIN_ID,
+                'policy': self.ROOT_ID
+            }
+        ]
+
+        self.assertEqual(response[1], 200)
+        self.assertEqual(response[0], target_response)
+
+    def test_get_members_400(self):
+        """Test case for 400 status response for GET requests using 'members' query parameter
+        """
+        with self.assertRaises(openapi_client.ApiException) as context:
+            self.api.get_role_with_http_info(
+                self.account,
+                NULL_BYTE,
+                'admin',
+                members=''
+            )
+
+        self.assertEqual(context.exception.status, 400)
+
+    def test_get_members_401(self):
+        """Test case for 401 status response for GET requests using 'members' query parameter
+        """
+        with self.assertRaises(openapi_client.ApiException) as context:
+            self.bad_auth_api.get_role_with_http_info(
+                self.account,
+                'user',
+                'admin',
+                members=''
+            )
+
+        self.assertEqual(context.exception.status, 401)
+
+    def test_get_members_404(self):
+        """Test case for 404 status response for GET requests using 'members' query parameter
+        """
+        with self.assertRaises(openapi_client.ApiException) as context:
+            self.api.get_role_with_http_info(
+                self.account,
+                'user',
+                'fakeUser',
+                members=''
+            )
+
+        self.assertEqual(context.exception.status, 404)
+
+    def test_get_members_422(self):
+        """Test case for 422 status response for GET requests using 'members' query parameter
+        """
+        with self.assertRaises(openapi_client.ApiException) as context:
+            self.api.get_role_with_http_info(
+                self.account,
+                'user',
+                'admin',
+                members='',
+                search=NULL_BYTE
+            )
+
+        self.assertEqual(context.exception.status, 422)
+
+    # Test cases confirming count, offset, limit and search query parameters
+
+    def test_get_members_count(self):
+        """Test using members & count query parameters on GET
+        Adds dev:user:bob as a member of group:userGroup, then queries for a member count
```

Comment:
I like adding this little explanation of how the test works in the doc-string! It probably makes it much more readable for someone who doesn't understand Conjur